types/module/module.go

```diff
@@ -299,40 +299,38 @@ func (m *Manager) EndBlock(ctx sdk.Ctx, req abci.RequestEndBlock) abci.ResponseE
 	defer sdk.TimeTrack(time.Now())
 	ctx = ctx.WithEventManager(sdk.NewEventManager())
 	validatorUpdates := []abci.ValidatorUpdate{}
-	UpdateToApply := &abci.ConsensusParams{}
+	var consensusUpdates *abci.ConsensusParams
 
 	for _, moduleName := range m.OrderEndBlockers {
+		// Use these validator updates if provided
 		moduleValUpdates := m.Modules[moduleName].EndBlock(ctx, req)
-
-		// use these validator updates if provided, the module manager assumes
-		// only one module will update the validator set
 		if len(moduleValUpdates) > 0 {
+			// the module manager assumes only one module will update the validator set
 			if len(validatorUpdates) > 0 {
 				panic("validator EndBlock updates already set by a previous module")
 			}
-
 			validatorUpdates = moduleValUpdates
 		}
-		//Currently its only a non empty struct on pocketcore module
+
+		//Currently its only a non-empty struct in the pocket-core modules
 		consensusParamUpdate := m.Modules[moduleName].ConsensusParamsUpdate(ctx)
 		if !consensusParamUpdate.Equal(&abci.ConsensusParams{}) {
-			UpdateToApply = consensusParamUpdate
+			consensusUpdates = consensusParamUpdate
 		}
 	}
 
-	//not adding empty struct  saves us from updating consensus params every block if there are no updates.
-	if UpdateToApply.Equal(&abci.ConsensusParams{}) {
+	// Only attach the consensus updates if present.
+	// Avoids sending the update (i.e. an empty struct) every block
+	if consensusUpdates != nil {
```

Comment:
NIT, I would just create a initialize a `ResponseEndBlock` and assign `Events` if `consensusUpdates != nil`.

```go

responseBlock := abci.ResponseEndBlock{
			ValidatorUpdates:      validatorUpdates,
			ConsensusParamUpdates: consensusUpdates,
		}

if consensusUpdates != nil {
responseBlock.ConsenusParamUpdates = consensusUpdates
}
return responseBlock
```